internal/vm/agent.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package vm
+
+import (
+	"net/url"
+)
+
+// IsAgentOnlyIO checks whether Stdout target is agent only in order to allow
+// log redirection entirely within the VM (currently cio.BinaryIO and cio.LogFile)
+func IsAgentOnlyIO(stdout string) bool {
+	parsed, err := url.Parse(stdout)
+	if err != nil {
```

Comment:
I'm guessing that most of the errors returned here are expected (`stdout` is just not a url), but should we at least log this error at Debug in case we get into a situation in the future where this starts returning unexpected errors?

Reply:
Probably wouldn't hurt. Done.